arch/arm/core/fault.c

```diff
@@ -524,4 +524,22 @@ void _FaultInit(void)
 #else
 #error Unknown ARM architecture
 #endif /* CONFIG_ARMV6_M_ARMV8_M_BASELINE */
+#if defined(CONFIG_BUILTIN_STACK_GUARD)
+	/* If Stack guarding via SP limit checking is enabled, disable
+	 * SP limit checking inside HardFault and NMI. This is done
+	 * in order to allow for the desired fault logging to execute
+	 * properly in all cases.
+	 *
+	 * Note that this could allow a Secure Firmware Main Stack
+	 * to descend into non-secure region during HardFault and
+	 * NMI exception entry. To prevent from this, non-secure
+	 * memory regions must be located higher than secure memory
+	 * regions.
```

Comment:
Can this be enforced with an assertion, or at least be documented outside of a comment buried inside `arch/arm`?

Reply:
Currently, there is no support for Zephyr builds that comprise of composite (Secure/Non-Secure) images; this comment is really looking into the future. However, when such images will be supported, we could document the rationale to allocate non-secure memory in region higher than secure memory, exactly at the place we will implement the allocation.